### PR TITLE
[15.0] account_chatter: Migration to v15.0 - T#55107

### PR DESCRIPTION
**Pull request by randall-vx**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,12 +2,15 @@
 __pycache__/
 *.py[cod]
 
+*.DS_Store*
```
Comment (luisg123v) [inline .gitignore:5]: Could you move this to a separated commit, please?

File: account_chatter/models/account_account.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    group_id = fields.Many2one(tracking=True)
+
+    def _message_track(self, tracked_fields, initial):
+        """Perform a field tracking over tax_ids and tag_ids
+
+        This is performed manually because field tracking over many2many fields is not
+        natively supported.
+        """
+        changes, tracking_value_ids = super()._message_track(tracked_fields, initial)
+        for display_name, display_info in tracked_fields.items():
```
Comment (luisg123v) [inline account_chatter/models/account_account.py]:
```suggestion
        for col_name, col_info in tracked_fields.items():
```

File: account_chatter/models/account_account.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            if display_name not in ('tag_ids', 'tax_ids'):
+                continue
+            initial_value = initial[display_name]
+            new_value = self[display_name]
+            if new_value != initial_value and (new_value or initial_value):
+                tracking_sequence = getattr(self._fields[display_name], 'track_sequence', 100)
```
Comment (luisg123v) [inline account_chatter/models/account_account.py]: Prioritize `tracking` attribute, as done [here](https://github.com/odoo/odoo/blob/e8882fb745d140eb8d4385b7a1d0f754c78453ab/addons/mail/models/models.py#L50)

File: account_chatter/models/account_account.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                initial_value = ", ".join(initial_value.mapped('name')) if initial_value else False
+                new_value = ", ".join(new_value.mapped('name')) if new_value else False
+                display_info['type'] = 'char'
+                tracking = self.env['mail.tracking.value'].create_tracking_values(
+                    initial_value, new_value, display_name, display_info, tracking_sequence)
```
Comment (luisg123v) [inline account_chatter/models/account_account.py]: Missing new parameter `_name`:

```suggestion
                    initial_value, new_value, display_name, display_info, tracking_sequence, self._name)
```

CC @rolandojduartem

File: account_chatter/README.rst [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/Vauxoo/addons-vauxoo/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed `feedback
+<https://github.com/vauxoo/
+addons-vauxoo/issues/new?body=module:%20
+account_chatter%0Aversion:%20
+12.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (luisg123v) [inline account_chatter/README.rst]:
```suggestion
15.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```

**Review by luisg123v — CHANGES_REQUESTED** [review]

File: account_chatter/models/account_account.py (hunk 4 above)
Comment (rolandojduartem) [reply account_chatter/models/account_account.py]: Maybe this could help https://github.com/odoo/odoo/blob/e8882fb745d140eb8d4385b7a1d0f754c78453ab/addons/mail/models/mail_tracking_value.py#L48

**Comment by randall-vx** [comment]
@luisg123v @rolandojduartem please check.

regards,

File: account_chatter/models/account_account.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    deprecated = fields.Boolean(tracking=True)
+    reconcile = fields.Boolean(tracking=True)
+    user_type_id = fields.Many2one(tracking=True)
+    tax_ids = fields.Many2many(tracking=True)
+    tag_ids = fields.Many2many(tracking=True)
+    group_id = fields.Many2one(tracking=True)
+
+    def _mail_track(self, tracked_fields, initial):
+        """ For a given record, fields to check (tuple column name, column info)
+        and initial values, return a valid command to create tracking values.
```
Comment (luisg123v) [inline account_chatter/models/account_account.py]: Previous docstring was OK, it described why the inheritance. New one is just a copy of the native one.

File: account_chatter/models/account_account.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            if new_value != initial_value and (new_value or initial_value):  # because browse null != False
+                tracking_sequence = getattr(
+                    # backward compatibility with old parameter name
+                    self._fields[col_name], 'tracking', getattr(self._fields[col_name], 'track_sequence', 100))
+                if tracking_sequence is True:
+                    tracking_sequence = 100
+                tracking = self.env['mail.tracking.value'].create_tracking_values(
+                    initial_value, new_value, col_name, col_info, tracking_sequence, self._name)
+                if tracking:
+                    if tracking['field_type'] == 'monetary':
```
Comment (luisg123v) [inline account_chatter/models/account_account.py]: Not necessary, as this inheritance is to add support to many2many fields

File: stock_by_warehouse/tests/test_stock_available_unreserved.py [hunk 8]
```diff
@@ -4,7 +4,7 @@
 class TestStockLogisticsWarehouse(TransactionCase):
 
     def setUp(self):
-        super(TestStockLogisticsWarehouse, self).setUp()
+        super().setUp()
```
Comment (luisg123v) [inline stock_by_warehouse/tests/test_stock_available_unreserved.py:7]: This is an unrelated change

**Review by luisg123v — CHANGES_REQUESTED** [review]

**Comment by luisg123v** [comment]
@randall-vx could you provide the task ID on PR's title, please?

File: account_chatter/models/account_account.py (hunk 6 above)
Comment (randall-vx) [reply account_chatter/models/account_account.py]: Done !

File: account_chatter/models/account_account.py (hunk 7 above)
Comment (randall-vx) [reply account_chatter/models/account_account.py]: @luisg123v could you improve your comment, please ?

File: stock_by_warehouse/tests/test_stock_available_unreserved.py (hunk 8 above)
Comment (randall-vx) [reply stock_by_warehouse/tests/test_stock_available_unreserved.py:7]: Done !

File: .gitignore (hunk 1 above)
Comment (randall-vx) [reply .gitignore:5]: Done !

**Comment by randall-vx** [comment]
@luisg123v could you review please ?

regards,

File: account_chatter/models/account_account.py (hunk 7 above)
Comment (luisg123v) [reply account_chatter/models/account_account.py]: This is in the original method because it's intended to support several field types, including monetary fields. Since you're extending that method to add value to a many2many field, this line is not required, because it's only applicable to monetary fields

**Comment by luisg123v** [comment]
@randall-vx only one comment is pending, other than that LGTM

File: account_chatter/models/account_account.py (hunk 7 above)
Comment (randall-vx) [reply account_chatter/models/account_account.py]: Done

**Review by luisg123v — APPROVED** [review]
